src/plugins/cosmos/components/modals/Send/hooks/useFormSend/useFormSend.tsx

```diff
@@ -0,0 +1,145 @@
+import { ExternalLinkIcon } from '@chakra-ui/icons'
+import { Link, Text, useToast } from '@chakra-ui/react'
+import { ChainAdapter } from '@shapeshiftoss/chain-adapters'
+import { chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+import { useTranslate } from 'react-polyglot'
+import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersProvider'
+import { useModal } from 'context/ModalProvider/ModalProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { ensLookup } from 'lib/ens'
+import { isEthAddress } from 'lib/utils'
+import { accountIdToUtxoParams } from 'state/slices/portfolioSlice/utils'
+
+import { SendInput } from 'components/Modals/Send/Form'
+
+export const useFormSend = () => {
+  const toast = useToast()
+  const translate = useTranslate()
+  const chainAdapterManager = useChainAdapters()
+  const { send } = useModal()
+  const {
+    state: { wallet }
+  } = useWallet()
+
+  const handleSend = async (data: SendInput) => {
+    if (wallet) {
+      try {
+        const adapter = chainAdapterManager.byChain(data.asset.chain)
+        const value = bnOrZero(data.cryptoAmount)
+          .times(bnOrZero(10).exponentiatedBy(data.asset.precision))
+          .toFixed(0)
+
+        const adapterType = adapter.getType()
+
+        let result
+
+        const { estimatedFees, feeType, address: to } = data
+        if (adapterType === ChainTypes.Ethereum) {
```

Comment:
`Remove all chain specific code from all files` -> all these Eth/ereum references look like chain-specific logic to me 👀